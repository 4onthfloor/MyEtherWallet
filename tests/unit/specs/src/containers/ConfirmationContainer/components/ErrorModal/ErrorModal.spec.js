import Vue from 'vue';
import { mount } from '@vue/test-utils';
import ErrorModal from '@/containers/ConfirmationContainer/components/ErrorModal/ErrorModal.vue';
import sinon from 'sinon';
import { Tooling } from '@@/helpers';

const hideModal = sinon.stub();
const showModal = sinon.stub();
const BModalStub = {
  name: 'b-modal',
  template: '<div><slot></slot></div>',
  props: ['to'],
  methods: {
    hide: hideModal,
    show: showModal
  }
};

const BBtnStub = {
  name: 'b-btn',
  template: '<div class="b-btn"><slot></slot></div>'
};

describe('ErrorModal.vue', () => {
  let localVue, i18n, wrapper, store;
  const message = 'message';
  const linkMessage = 'linkMessage';
  const linkTo = '/linkTo';

  const spy = sinon.stub();
  const mockRoute = {
    push: spy
  };

  beforeAll(() => {
    const baseSetup = Tooling.createLocalVueInstance();
    localVue = baseSetup.localVue;
    i18n = baseSetup.i18n;
    store = baseSetup.store;

    Vue.config.warnHandler = () => {};
  });

  beforeEach(() => {
    wrapper = mount(ErrorModal, {
      localVue,
      i18n,
      store,
      attachToDocument: true,
      propsData: { message, linkTo },
      stubs: {
        'b-modal': BModalStub,
        'b-btn': BBtnStub
      },
      mocks: {
        $router: mockRoute
      }
    });
  });

  afterAll(() => {
    wrapper.destroy();
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correct message props', () => {
    const message = wrapper.find('.d-block p').text();
    expect(message.trim()).toEqual(message);
  });

  it('should render correct linkMessage props', async () => {
    wrapper.vm.$refs.errorModal.show();
    await wrapper.vm.$nextTick();
    const container = wrapper.findAll('.button-container');
    expect(container.at(0).text()).toEqual('An Error Occured');
    wrapper.setProps({ linkMessage });
    expect(container.at(0).text()).toEqual(linkMessage);
  });

  describe('ErrorModal.vue Methods', () => {
    it('should render correct hideModal method', () => {
      wrapper.vm.hideModal();
      expect(hideModal.called).toBe(true);
    });
  });

  it('should dismount properly', () => {
    expect(wrapper.destroy()).toBe(undefined);
  });
});
