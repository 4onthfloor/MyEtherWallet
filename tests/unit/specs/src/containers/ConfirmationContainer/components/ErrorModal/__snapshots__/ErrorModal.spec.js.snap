// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorModal.vue renders correctly 1`] = `
<div centered="" hide-footer="" hide-header="" static="" lazy="" class="bootstrap-modal">
  <div class="d-block text-center"><i aria-hidden="true" class="check-icon fa fa-close"></i>
    <h2 class="title">Error</h2>
    <p>message</p>
  </div>
  <div class="button-container">
    <div class="b-btn mid-round-button-lightgrey-filled close-button">
      An Error Occured
    </div>
  </div>
</div>
`;
